File: mobile_apps/ios_app/TigerWallet/Screens/NFTScreen.swift
import SwiftUI

// NFT Screen - NFT Gallery
struct NFTScreen: View {
    @State private var selectedTab: String = "Collectibles"
    @State private var searchText: String = ""
    
    let tabs = ["Collectibles", "Activity", "OpenSea"]
    
    let sampleNFTs = [
        NFTItem(name: "Bored Ape #1234", collection: "Bored Ape Yacht Club", image: "🦍", price: "45.5 ETH"),
        NFTItem(name: "CryptoPunk #5678", collection: "CryptoPunks", image: "👾", price: "32.0 ETH"),
        NFTItem(name: "Azuki #9012", collection: "Azuki", image: "🥷", price: "15.2 ETH"),
        NFTItem(name: "Doodle #3456", collection: "Doodles", image: "🎨", price: "3.5 ETH"),
        NFTItem(name: "Moonbird #7890", collection: "Moonbirds", image: "🐦", price: "8.1 ETH")
    ]
    
    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Search Bar
                HStack {
                    Image(systemName: "magnifyingglass")
                        .foregroundColor(.secondary)
                    TextField("Search NFTs", text: $searchText)
                }
                .padding()
                .background(Color.gray.opacity(0.1))
                .cornerRadius(10)
                .padding()
                
                // Tab Selector
                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 16) {
                        ForEach(tabs, id: \.self) { tab in
                            Button(action: { selectedTab = tab }) {
                                VStack {
                                    Text(tab)
                                        .fontWeight(selectedTab == tab ? .bold : .regular)
                                    Rectangle()
                                        .fill(selectedTab == tab ? Color.orange : Color.clear)
                                        .frame(height: 2)
                                }
                            }
                            .foregroundColor(selectedTab == tab ? .primary : .secondary)
                        }
                    }
                    .padding(.horizontal)
                }
                
                if selectedTab == "Collectibles" {
                    // NFT Grid
                    ScrollView {
                        LazyVGrid(columns: [
                            GridItem(.flexible()),
                            GridItem(.flexible())
                        ], spacing: 16) {
                            ForEach(filteredNFTs) { nft in
                                nftCard(nft: nft)
                            }
                        }
                        .padding()
                    }
                } else if selectedTab == "Activity" {
                    activityView
                } else {
                    openSeaView
                }
            }
            .navigationTitle("NFTs")
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: {}) {
                        Image(systemName: "square.and.arrow.down")
                    }
                }
            }
        }
    }
    
    var filteredNFTs: [NFTItem] {
        if searchText.isEmpty {
            return sampleNFTs
        }
        return sampleNFTs.filter { nft in
            nft.name.localizedCaseInsensitiveContains(searchText) ||
            nft.collection.localizedCaseInsensitiveContains(searchText)
        }
    }
    
    func nftCard(nft: NFTItem) -> some View {
        VStack(alignment: .leading, spacing: 8) {
            // NFT Image placeholder
            ZStack {
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.gray.opacity(0.2))
                    .aspectRatio(1, contentMode: .fit)
                
                Text(nft.image)
                    .font(.system(size: 60))
            }
            
            VStack(alignment: .leading, spacing: 4) {
                Text(nft.name)
                    .font(.caption)
                    .fontWeight(.semibold)
                    .lineLimit(1)
                
                Text(nft.collection)
                    .font(.caption2)
                    .foregroundColor(.secondary)
                    .lineLimit(1)
                
                Text(nft.price)
                    .font(.caption)
                    .foregroundColor(.orange)
            }
        }
        .padding(8)
        .background(Color.gray.opacity(0.1))
        .cornerRadius(12)
    }
    
    var activityView: some View {
        ScrollView {
            VStack(spacing: 12) {
                ForEach(0..<5, id: \.self) { _ in
                    HStack {
                        Image(systemName: "arrow.up.right")
                            .foregroundColor(.red)
                        VStack(alignment: .leading) {
                            Text("Sent")
                                .font(.subheadline)
                            Text("Bored Ape #1234")
                                .font(.caption)
                                .foregroundColor(.secondary)
                        }
                        Spacer()
                        Text("- 1 NFT")
                            .font(.caption)
                    }
                    .padding()
                    .background(Color.gray.opacity(0.1))
                    .cornerRadius(8)
                }
            }
            .padding()
        }
    }
    
    var openSeaView: some View {
        VStack(spacing: 20) {
            Image(systemName: "globe")
                .font(.system(size: 60))
                .foregroundColor(.secondary)
            
            Text("OpenSea Integration")
                .font(.headline)
            
            Text("Connect to OpenSea to view and trade NFTs")
                .font(.subheadline)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)
            
            Button(action: {}) {
                Text("Connect OpenSea")
                    .padding()
                    .background(Color.orange)
                    .foregroundColor(.white)
                    .cornerRadius(12)
            }
        }
        .padding()
    }
}

struct NFTItem: Identifiable {
    let id = UUID()
    let name: String
    let collection: String
    let image: String
    let price: String
}

#Preview {
    NFTScreen()
}

File: mobile_apps/android_app/TigerWallet/app/src/main/java/com/tigerwallet/app/ui/screens/NFTScreen.kt
package com.tigerwallet.app.ui.screens

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.grid.GridCells
import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
import androidx.compose.foundation.lazy.grid.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp

// NFT Screen - NFT Gallery
class NFTScreen : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            MaterialTheme {
                NFTScreenContent()
            }
        }
    }
}

@OptIn(ExperimentalMaterialApi::class)
@Composable
fun NFTScreenContent() {
    var selectedTab by remember { mutableStateOf("Collectibles") }
    var searchText by remember { mutableStateOf("") }
    
    val tabs = listOf("Collectibles", "Activity", "OpenSea")
    val nfts = listOf(
        NFTData("Bored Ape #1234", "Bored Ape Yacht Club", "🦍", "45.5 ETH"),
        NFTData("CryptoPunk #5678", "CryptoPunks", "👾", "32.0 ETH"),
        NFTData("Azuki #9012", "Azuki", "🥷", "15.2 ETH"),
        NFTData("Doodle #3456", "Doodles", "🎨", "3.5 ETH"),
        NFTData("Moonbird #7890", "Moonbirds", "🐦", "8.1 ETH"),
        NFTData("Pudgy #2345", "Pudgy Penguins", "🐧", "2.8 ETH")
    )

    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("NFTs") },
                actions = {
                    IconButton(onClick = { /* Download */ }) {
                        Icon(androidx.compose.material.icons.Icons.Default.Download, contentDescription = "Download")
                    }
                }
            )
        }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
        ) {
            // Search Bar
            OutlinedTextField(
                value = searchText,
                onValueChange = { searchText = it },
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 16.dp, vertical = 8.dp),
                placeholder = { Text("Search NFTs") },
                leadingIcon = { Icon(androidx.compose.material.icons.Icons.Default.Search, contentDescription = null) },
                singleLine = true,
                shape = RoundedCornerShape(12.dp)
            )

            // Tab Selector
            TabRow(selectedTabIndex = tabs.indexOf(selectedTab)) {
                tabs.forEachIndexed { index, tab ->
                    Tab(
                        selected = selectedTab == tab,
                        onClick = { selectedTab = tab },
                        text = { Text(tab) }
                    )
                }
            }

            when (selectedTab) {
                "Collectibles" -> nftGrid(nfts, searchText)
                "Activity" -> activityView()
                "OpenSea" -> openSeaView()
            }
        }
    }
}

@Composable
fun nftGrid(nfts: List<NFTData>, search: String) {
    val filtered = if (search.isEmpty()) nfts else nfts.filter {
        it.name.contains(search, ignoreCase = true) || it.collection.contains(search, ignoreCase = true)
    }

    LazyVerticalGrid(
        columns = GridCells.Fixed(2),
        contentPadding = PaddingValues(16.dp),
        horizontalArrangement = Arrangement.spacedBy(12.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        items(filtered) { nft ->
            NFTCard(nft = nft)
        }
    }
}

@Composable
fun NFTCard(nft: NFTData) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp)
    ) {
        Column {
            // NFT Image placeholder
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .aspectRatio(1f)
                    .background(MaterialTheme.colors.surfaceVariant),
                contentAlignment = Alignment.Center
            ) {
                Text(nft.image, style = MaterialTheme.typography.h2)
            }
            
            Column(modifier = Modifier.padding(12.dp)) {
                Text(
                    nft.name,
                    style = MaterialTheme.typography.subtitle2,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
                Text(
                    nft.collection,
                    style = MaterialTheme.typography.caption,
                    color = MaterialTheme.colors.onSurfaceVariant,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
                Spacer(modifier = Modifier.height(4.dp))
                Text(
                    nft.price,
                    style = MaterialTheme.typography.caption,
                    color = MaterialTheme.colors.primary
                )
            }
        }
    }
}

@Composable
fun activityView() {
    val activities = listOf(
        "Sent Bored Ape #1234" to "- 1 NFT",
        "Received CryptoPunk #5678" to "+ 1 NFT",
        "Listed Azuki #9012" to "2.5 ETH",
        "Sold Doodle #3456" to "3.5 ETH"
    )

    Column(
        modifier = Modifier.fillMaxSize().padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        activities.forEach { (title, amount) ->
            Card(modifier = Modifier.fillMaxWidth()) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column {
                        Text(title, style = MaterialTheme.typography.subtitle2)
                        Text("Just now", style = MaterialTheme.typography.caption, color = MaterialTheme.colors.onSurfaceVariant)
                    }
                    Text(amount, style = MaterialTheme.typography.subtitle2, color = MaterialTheme.colors.primary)
                }
            }
        }
    }
}

@Composable
fun openSeaView() {
    Column(
        modifier = Modifier.fillMaxSize().padding(32.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center
    ) {
        Icon(
            androidx.compose.material.icons.Icons.Default.Language,
            contentDescription = null,
            modifier = Modifier.size(80.dp),
            tint = MaterialTheme.colors.onSurfaceVariant
        )
        Spacer(modifier = Modifier.height(16.dp))
        Text("OpenSea Integration", style = MaterialTheme.typography.h6)
        Spacer(modifier = Modifier.height(8.dp))
        Text(
            "Connect to OpenSea to view and trade NFTs",
            style = MaterialTheme.typography.body2,
            color = MaterialTheme.colors.onSurfaceVariant
        )
        Spacer(modifier = Modifier.height(24.dp))
        Button(onClick = { /* Connect */ }) {
            Text("Connect OpenSea")
        }
    }
}

data class NFTData(
    val name: String,
    val collection: String,
    val image: String,
    val price: String
)
